pkg/operator/ceph/controller/cluster_info.go

```diff
@@ -166,7 +177,33 @@ func CreateOrLoadClusterInfo(clusterdContext *clusterd.Context, context context.
 		clusterInfo.CephCred.Username = string(secret.Data["userID"])
 		clusterInfo.CephCred.Secret = string(secret.Data["userKey"])
 	}
```

Comment:
To truly maintain backward compatibility, let's go with the following approach.
1. Only the rook operator code should be touched by the changes. 
2. No example yamls such as the toolbox or toolbox job should be updated. They will keep using the `client.admin` keyring
3. Here where we update the secret (applies to upgraded clusters):
   a. If it's an external cluster, copy the keyring to the new key name as you already have (lines 181-184)
   b. If not external, generate the `rookoperator` keyring as you already have (lines 186-201)
4. New clusters will need to create the secret differently (lines 353-356 below):
   a. Set the legacy username and keyring to `client.admin`
   b. Set the new username and keyring to `client.admin`. The new keyring `rookoperator` will **not** be created or used until **after the mons** are initialized. 
5. After the mons are initialized (likely in the `postMonStartupActions()` method):
   a. If using the `client.admin` keyring, generate the `rookoperator` keyring, just the same as an upgraded cluster. Hopefully this is as simple as calling this method `CreateOrLoadClusterInfo()` again and it should trigger the upgrade path (lines 186-201).
 
This will allow existing toolbox, external scripts, and so on to continue working with the existing admin keyring and using the same key name in the secret. It will be a bit awkward that we upgrade the keyring even in a new cluster, but it does seem necessary for fully compatibility.